### DISCUSSION TITLE
How to calculate de right weight for a target - Kong upstream

### DISCUSSION THREAD

**Discussion by VitorNilson** [Kong/kong#12085] [Help]
I'm currently creating an upstream for a service wich has a "stable" version and a "canary deployment" version.
I have to send at least 80% of the traffic to the stable and up to 20% to canary.

The [docummentation](https://legacy-gateway--kongdocs.netlify.app/gateway-oss/0.10.x/loadbalancing/#upstream) suggests that we should use at least 100 slots per target and as much higher the slots are as much better is the randomness, but the tradeoff is as higher the number of slots as expensive the addition/removal of targets.

So here's my first question: 
What can Happen if I have a big slots size and I need to remove one of my targets (in my case I would remove the canary)? And how to calculate the right amount of slots?

Going further: I have created an upstream with 1000 slots.

How can I calculate the right weight of each target so, in my case, 80% of the traffic gets redirected to "stable" and 20% to "canary"?

Its simple as this?:

stable: weight = 800
canary: weight = 200

**Comment by chobits** [comment]
> Its simple as this?:
> stable: weight = 800
> canary: weight = 200

Yes. 1000 slots are available. The more slots you configure, the more precise the traffic balancing become. 
However, configuring 100 slots is sufficient to maintain an error margin of 1%.
